electron/js/menu/TrayIconHandler.js

```diff
@@ -0,0 +1,85 @@
+const {app, Menu, nativeImage, Tray} = require('electron');
+const config = require('./../config');
+const environment = require('./../environment');
+const lifecycle = require('./../lifecycle');
+const locale = require('./../../locale/locale');
+const path = require('path');
+const windowManager = require('./../window-manager');
+
+class TrayIconHandler {
+  constructor(appIcon = new Tray(nativeImage.createEmpty())) {
+    this.appIcon = appIcon;
+    this.icons = this.createIcons();
+    this.lastUnreadCount = 0;
+    this.initTrayIcon();
+  }
+
+  initTrayIcon() {
+    this.appIcon.setImage(this.icons.tray);
+    if (this.hasTrayMenuSupport) {
+      this.buildTrayMenu();
+    }
+  }
+
+  get defaultImageExtension() {
+    return environment.platform.IS_WINDOWS ? 'ico' : 'png';
+  }
+
+  get hasTrayMenuSupport() {
+    return !environment.platform.IS_MAC_OS;
+  }
+
+  get hasOverlaySupport() {
+    return environment.platform.IS_WINDOWS;
+  }
+
+  buildTrayMenu() {
+    const contextMenu = Menu.buildFromTemplate([
+      {
+        click: () => windowManager.showPrimaryWindow(),
+        label: locale.getText('trayOpen'),
+      },
+      {
+        click: async () => await lifecycle.quit(),
+        label: locale.getText('trayQuit'),
+      },
+    ]);
+
+    this.appIcon.on('click', () => windowManager.showPrimaryWindow());
+    this.appIcon.setContextMenu(contextMenu);
+    this.appIcon.setToolTip(config.NAME);
+  }
+
+  createIcons() {
```

Comment:
No need to expose this outside of the constructor

Reply:
This one doesn't need `this` at all, so it could definitely be defined in the private scope